save-preprocessor/src/main/kotlin/org/cqfn/save/preprocessor/utils/FileUtil.kt

```diff
@@ -0,0 +1,10 @@
+package org.cqfn.save.preprocessor.utils
+
+import java.io.File
+import java.math.BigInteger
+import java.security.MessageDigest
+
+fun File.toHash(): String {
+    val md = MessageDigest.getInstance("SHA-256")
+    return BigInteger(1, md.digest(this.readBytes())).toString(16)
```

Comment:
I suggest something like [this](https://stackoverflow.com/questions/304268/getting-a-files-md5-checksum-in-java) in order not to read the whole file into memory